Optimize client-side dead reckoning

axblk opened the pull request:
- Use a client-side evolve limit (3 sec, same as server-side) to prevent the client from freezing

- Reuse the evolved character from the last snapshot. Instead of up to 300 evolve calls (3*50 ticks for each Cur and Prev), this usually needs 1 or 2 per character in a snapshot. This is based on https://github.com/ddnet/ddnet/pull/1964

heinrich5991 commented:
Wrong place to limit it IMO. This is a general purpose function inside the client, could be used for other stuff in the future. Should be done directly when receiving instead, if possible.

axblk commented:
@heinrich5991 you are right. I moved it to `CGameClient::OnNewSnapshot()` and also added the functionality from https://github.com/ddnet/ddnet/pull/1964
